src/runtime_src/xdp/profile/database/static_info/xclbin_info.cpp

```diff
@@ -94,17 +145,568 @@ namespace xdp {
     for (auto i : nocList) {
       delete i ;
     }
   }
 
+  XclbinInfo::XclbinInfo(XclbinInfoType xclbinType) : type(xclbinType)
+  {
+      if(xclbinType == XclbinInfoType::XCLBIN_PL_ONLY) {
+        pl.valid  = true;
+        aie.valid = false;
+      } else if(xclbinType == XclbinInfoType::XCLBIN_AIE_ONLY) {
+        pl.valid  = false;
+        aie.valid = true;
+      }
+  }
+
   XclbinInfo::~XclbinInfo()
   {
     if (deviceIntf)
       delete deviceIntf ;
   }
 
+  ConfigInfo::ConfigInfo() : type(CONFIG_AIE_PL)
+  {
+
+  }
+
+  ConfigInfo::ConfigInfo(XclbinInfo* xclbin) : type(CONFIG_AIE_PL)
+  {
+    currentXclbins.push_back(xclbin);
+  }
+
+  ConfigInfo::~ConfigInfo()
+  {
+    for(auto xclbin : currentXclbins) {
+      delete xclbin;
+    }
+    currentXclbins.clear();
+  }
+
+  xrt_core::uuid ConfigInfo::getConfigUuid()
+  {
+    if(currentXclbins.size()==1)
+      return currentXclbins.back()->uuid;
+
+    std::string mix_uuid_str;
+    for(auto xclbin : currentXclbins)
+      mix_uuid_str += xclbin->uuid.to_string();
+    
+    return xrt_core::uuid(mix_uuid_str);
+  }
+
+  std::vector<xrt_core::uuid> ConfigInfo::getConfigUuids()
+  {
+    std::vector<xrt_core::uuid> uuids;
+    for(auto &xclbin : currentXclbins)
+      uuids.push_back(xclbin->uuid);
+    
+    return uuids;
+  }
+
+  void ConfigInfo::addXclbin(XclbinInfo* xclbin)
+  {
+    currentXclbins.push_back(xclbin);
+  }
+
+  void ConfigInfo::updateType(ConfigInfoType cfgType)
+  {
+    type = cfgType;
+  }
+
+  bool ConfigInfo::containsXclbin(xrt_core::uuid& uuid)
+  {
+    for(auto &xclbin : currentXclbins)
+    {
+      if(xclbin->uuid == uuid)
+        return true;
+    }
+    
+    return false;
+  }
+
+  XclbinInfo* ConfigInfo::getPlXclbin()
+  {
+    for(auto xclbin : currentXclbins)
+    {
+      if(xclbin->pl.valid)
+        return xclbin;
+    }
+    return nullptr;
+  }
+
+  XclbinInfo* ConfigInfo::getAieXclbin()
+  {
+    for(auto xclbin : currentXclbins)
+    {
+      if(xclbin->aie.valid)
+        return xclbin;
+    }
+    return nullptr;
+  }
+
+  std::string ConfigInfo::getXclbinNames()
+  {
+    std::string name = "";
+    if(!currentXclbins.empty()) {
+      name += currentXclbins.front()->name;
+      for(size_t i=1; i<currentXclbins.size(); i++) {
+        name += ", " + currentXclbins[i]->name;
+      }
+    }
+    return name ;
+  }
+
+  bool ConfigInfo::isAiePlusPl()
+  {
+    if (type == CONFIG_AIE_PL || type == CONFIG_AIE_PL_FORMED)
+      return true;
+    return false; 
+  }
+  
+  bool ConfigInfo::isAieOnly()
+  {
+    return type == CONFIG_AIE_ONLY;
+  }
+
+  bool ConfigInfo::isPlOnly()
+  {
+    return type == CONFIG_PL_ONLY;
+  }
+
+  bool ConfigInfo::hasXclbin(XclbinInfo* xclbin)
+  {
+    for(auto bin : currentXclbins) {
+      if(bin == xclbin)
+        return true;
+    }
+    return false;
+  }
+
+    bool ConfigInfo::hasFloatingAIMWithTrace(XclbinInfo* xclbin)
+    {
+      for(auto bin : currentXclbins)
+        if(bin == xclbin)
```

Comment:
Do we need a check here to confirm that the xclbin in arguments is pl xclbin? This comment is for all the functions